controllers/nnf_clientmount_controller.go

```diff
@@ -196,44 +197,82 @@ func (r *NnfClientMountReconciler) changeMountAll(ctx context.Context, clientMou
 }
 
 // changeMount mount or unmounts a single mount point described in the ClientMountInfo object
-func (r *NnfClientMountReconciler) changeMount(ctx context.Context, clientMountInfo dwsv1alpha1.ClientMountInfo, mount bool, log logr.Logger) error {
+func (r *NnfClientMountReconciler) changeMount(ctx context.Context, clientMountInfo dwsv1alpha1.ClientMountInfo, shouldMount bool, log logr.Logger) error {
 
-	if clientMountInfo.Device.Type != dwsv1alpha1.ClientMountDeviceTypeReference {
-		return dwsv1alpha1.NewResourceError(fmt.Sprintf("Invalid device type %s", clientMountInfo.Device.Type), nil).WithFatal()
-	}
+	switch clientMountInfo.Device.Type {
+	case dwsv1alpha1.ClientMountDeviceTypeLustre:
+		mountPath := clientMountInfo.MountPath
 
-	if clientMountInfo.Device.DeviceReference.ObjectReference.Kind != reflect.TypeOf(nnfv1alpha1.NnfNodeStorage{}).Name() {
-		return dwsv1alpha1.NewResourceError(fmt.Sprintf("Invalid device reference kind %s", clientMountInfo.Device.DeviceReference.ObjectReference.Kind), nil).WithFatal()
-	}
+		_, testEnv := os.LookupEnv("NNF_TEST_ENVIRONMENT")
 
-	namespacedName := types.NamespacedName{
-		Name:      clientMountInfo.Device.DeviceReference.ObjectReference.Name,
-		Namespace: clientMountInfo.Device.DeviceReference.ObjectReference.Namespace,
-	}
+		var mounter mount.Interface
+		if testEnv {
+			mounter = mount.NewFakeMounter([]mount.MountPoint{})
+		} else {
+			mounter = mount.New("")
+		}
 
-	nodeStorage := &nnfv1alpha1.NnfNodeStorage{}
-	if err := r.Get(ctx, namespacedName, nodeStorage); err != nil {
-		return err
-	}
+		isNotMountPoint, _ := mount.IsNotMountPoint(mounter, mountPath)
 
-	allocationStatus := nodeStorage.Status.Allocations[clientMountInfo.Device.DeviceReference.Data]
-	fileShare, err := r.getFileShare(allocationStatus.FileShare.ID, allocationStatus.FileSystem.ID)
-	if err != nil {
-		return dwsv1alpha1.NewResourceError("Could not get file share", err).WithFatal()
-	}
+		if shouldMount {
+			if isNotMountPoint {
 
-	if mount {
-		fileShare.FileSharePath = clientMountInfo.MountPath
-	} else {
-		fileShare.FileSharePath = ""
-	}
+				mountSource := clientMountInfo.Device.Lustre.MgsAddresses +
+					":/" +
+					clientMountInfo.Device.Lustre.FileSystemName
+
+				if !testEnv {
+					if err := os.MkdirAll(mountPath, 0755); err != nil {
```

Comment:
When this os.MkdirAll() or mounter.Mount() fail, the error recorded in the NnfAccess is this:

```
status:
  error:
    debugMessage: 'Mount/Unmount failed: mkdir : no such file or directory'
    recoverable: true
  ready: false
  state: mounted
```

To help with debugging, I think we should get the value of mountPath into that message.


The message that shows up in the log is this:
```
2022-09-02T15:18:28.551-0500	INFO	controllers.NnfClientMount	Mount/Unmount failed: mkdir : no such file or directory	{"ClientMount": "rabbit-dev-01/default-on-rabbit"}
```

I got here by testing clientmount on its own, constructing my own (incorrect, oops) NnfAccess to drive it.